opentracing/propagation.go

```diff
@@ -0,0 +1,299 @@
+		default:
+			return nil, fmt.Errorf("Unknown TracerState field: %v", k)
+		}
+	}
+
+	if requiredFieldCount < 3 {
+		return nil, fmt.Errorf("Only found %v of 3 required fields", requiredFieldCount)
+	}
+
+	sp := newAppdashSpan(operationName, p.tracer)
+	sp.Recorder = p.tracer.newChildRecorder(propagatedSpanID, traceID)
+	sp.sampled = sampled
+	sp.attributes = splitTextCarrier.TraceAttributes
+	sp.tags = make(map[string]interface{}, 0)
+
+	return sp, nil
+}
+
+func (p *splitBinaryPropagator) InjectSpan(
+	sp opentracing.Span,
+	carrier interface{},
+) error {
+	sc := sp.(*Span)
+	splitBinaryCarrier, ok := carrier.(*opentracing.SplitBinaryCarrier)
+	if !ok {
+		return opentracing.ErrInvalidCarrier
+	}
+	var err error
+	var sampledByte byte
+	if sc.sampled {
+		sampledByte = 1
+	}
+
+	// Handle the trace and span ids, and sampled status.
+	contextBuf := new(bytes.Buffer)
+	err = binary.Write(contextBuf, binary.BigEndian, uint64(sc.Recorder.SpanID.Trace))
+	if err != nil {
+		return err
+	}
+
+	err = binary.Write(contextBuf, binary.BigEndian, uint64(sc.Recorder.SpanID.Span))
+	if err != nil {
+		return err
+	}
+
+	err = binary.Write(contextBuf, binary.BigEndian, sampledByte)
+	if err != nil {
+		return err
+	}
+
+	// Handle the attributes.
+	attrsBuf := new(bytes.Buffer)
+	err = binary.Write(attrsBuf, binary.BigEndian, int32(len(sc.attributes)))
+	if err != nil {
+		return err
+	}
+
+	sc.Lock()
+	for k, v := range sc.attributes {
+		keyBytes := []byte(k)
+		lenKeyBytes := int32(len(keyBytes))
+		if err := binary.Write(attrsBuf, binary.BigEndian, lenKeyBytes); err != nil {
+			return err
+		}
+		if err := binary.Write(attrsBuf, binary.BigEndian, keyBytes); err != nil {
+			return err
+		}
+
+		valBytes := []byte(v)
+		lenValBytes := int32(len(valBytes))
+		if err := binary.Write(attrsBuf, binary.BigEndian, lenValBytes); err != nil {
+			return err
+		}
+		if err := binary.Write(attrsBuf, binary.BigEndian, valBytes); err != nil {
+			return err
+		}
+	}
+	sc.Unlock()
+
+	splitBinaryCarrier.TracerState = contextBuf.Bytes()
+	splitBinaryCarrier.TraceAttributes = attrsBuf.Bytes()
+	return nil
+}
+
+func (p *splitBinaryPropagator) JoinTrace(
+	operationName string,
+	carrier interface{},
+) (opentracing.Span, error) {
+	var err error
+	splitBinaryCarrier, ok := carrier.(*opentracing.SplitBinaryCarrier)
+	if !ok {
+		return nil, opentracing.ErrInvalidCarrier
+	}
+
+	// Handle the trace, span ids, and sampled status.
+	contextReader := bytes.NewReader(splitBinaryCarrier.TracerState)
+	var traceID, propagatedSpanID uint64
+	var sampledByte byte
+
+	err = binary.Read(contextReader, binary.BigEndian, &traceID)
+	if err != nil {
+		return nil, opentracing.ErrTraceCorrupted
+	}
+	err = binary.Read(contextReader, binary.BigEndian, &propagatedSpanID)
+	if err != nil {
+		return nil, opentracing.ErrTraceCorrupted
+	}
+	err = binary.Read(contextReader, binary.BigEndian, &sampledByte)
+	if err != nil {
+		return nil, opentracing.ErrTraceCorrupted
+	}
```

Comment:
Not at all, in fact it would encode and decode the same exact binary sequence. A struct doesn't add any special encoding of any sort, for example check out this test I just wrote: http://play.golang.org/p/61Pi-UmsLc